service/internal/di/service_container.go

```diff
@@ -114,19 +117,49 @@ func BuildServiceContainer() container.Container {
 		return service
 	})
 
+	// ctr.Singleton(func(
+		// conesearchService *conesearch.ConesearchService,
+		// metadataService *metadata.MetadataService,
+		// config *config.Config,
+	// ) *httpservice.HttpServer {
+		// server, err := httpservice.NewHttpServer(conesearchService, metadataService, config.Service)
+		// if err != nil {
+			// panic(fmt.Errorf("Could not register HttpServer: %w", err))
+		// }
+		// if server == nil {
+			// panic("Server nil while registering HttpServer")
+		// }
+		// return server
+	// })
+
+	ctr.Singleton(func(
+		conesearchService *conesearch.ConesearchService,
+		metadataService *metadata.MetadataService,
+		config *config.Config,
+	) *api.API {
+		api, err := api.New(conesearchService, metadataService, config.Service)
+		if err != nil {
+			panic(fmt.Errorf("Could not register API: %w", err))
+		}
+		if api == nil {
+			panic("api nil while registering API")
+		}
```

Comment:
podría ser `if api == nil || err != nil`

```suggestion
		if err != nil || api == nil {
			panic(fmt.Errorf("Could not register API: %w", err))
		}
```

De todas formas me queda la duda de por qué esos dos casos y no solo revisar el `err`